test/framework/src/main/java/org/elasticsearch/cli/TestSystemTerminal.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cli;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+
+public class TestSystemTerminal extends Terminal {
```

Comment:
We already have a MockTerminal. Can we change that to be suitable for the tests that need this?

Reply:
I added this because there was a bug in `Terminal.SystemTerminal` where it created a new `BufferedReader` every time we tried to read a line, making it impossible to read multiline input. Since we write the code for SystemTerminal and need to maintain it, it seems appropriate to have tests for SystemTerminal behavior. _However_, if I make a few things visible for testing, I can test `SystemTerminal#getReader` directly and avoid making mock classes and testing indirectly via behavior.

tl;dr, thanks for drawing attention to this, and I can get rid of it.